### ISSUE TITLE
MM32SPIN05PF

### ISSUE THREAD

**Issue by BigVoo** [AILIFE4798/Hoverboard-Firmware-Hack-Gen2.x-MM32#19]
I was able to run the pinfinder and the main firmwares on both the master and slave boards of MM32SPIN05PF
.  Wheels spin and everything, but when I try to run the TESTSPEED.ino on the ESP32 connected to the boards, I get nothing.  I changed the battery voltage to zero on both boards, but still not seeing anything in the Serial Monitor.  I've made all of the changes in Arduino to run the ESP32.  Any suggestions?

**Comment by RoboDurden** [comment]
I am not the coder of this nice firmware but i made a youtube tutorial on it, did you watch it ?

https://www.youtube.com/watch?v=wUR05iwLMtM

please share the link to the .ino code you installed on your esp32 and quote the cide changes you have made here.

**Comment by BigVoo** [comment]
https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x-GD32/tree/main/Arduino%20Examples/TestSpeed

![testspeedchanges](https://github.com/user-attachments/assets/8723f6c4-057b-4f9d-99de-91dc6070a012)

**Comment by RoboDurden** [comment]
Yes, my hoverserial with my uartBus protocol used to work with the MM32 firmware.
Did you also use my autodetect.ino to run pinFinder ?
The idea with autodetect.ino is that you do not need to change any pin connections after installing the final firmware.

Maybe try https://github.com/reed-dan/hoverboard_hack_esp32_manualspeed, as AiLife does link this as the code example on his repo: https://gitlab.com/ailife8881/Hoverboard-Firmware-Hack-Gen2.x-MM32

**Comment by RoboDurden** [comment]
@AILIFE4798 , i have forgotten most of your MM32 workflow.
Maybe people succeed with the nice UI of pinfinder but run into trouble with the final firmware.
There should be some feedback that tells the user that the pin definitions from pinFinder were successfully found. Some startup melody. And then maybe some beeping error code when no serial commands get received..

**Comment by I-hate-2FA** [comment]
i have, if valid config is found itll boot with fast 5 beep, if no valid config is found, slow 1s beep and all led flashing, led display battery voltage in normal mode, flashing red is enter protection

**Comment by BigVoo** [comment]
> Yes, my hoverserial with my uartBus protocol used to work with the MM32 firmware. Did you also use my autodetect.ino to run pinFinder ? The idea with autodetect.ino is that you do not need to change any pin connections after installing the final firmware.
> 
> Maybe try https://github.com/reed-dan/hoverboard_hack_esp32_manualspeed, as AiLife does link this as the code example on his repo: https://gitlab.com/ailife8881/Hoverboard-Firmware-Hack-Gen2.x-MM32

I used the autodetect, but I will run everything again and check for the led and beeps.

**Comment by BigVoo** [comment]
@RoboDurden, Can you explain how to change the default mode live over remote uartbus?  I finally have both wheels spinning and was able to modify the code to have the wheels change via a remote control, but the wheel movement is not fluid.  Any help will be appreciated.

**Comment by BigVoo** [comment]
ok, so is it best to reflash pinmode and change the idrivmode there or is it better to implement your commands and push it through using Putty?  Or does it matter?

**Comment by RoboDurden** [comment]
if you have putty ready to send M1 to ESP32, i would try that first.
If nothing happens, then see if the pinFinder firmware can change drive mode.

**Comment by I-hate-2FA** [comment]
default is already drivemode 1
